### PR TITLE
[path_provider_linux] Migrate to nnbd

### PR DESCRIPTION
**Pull request by gaetschwartz**
Migrating `path_provider_linux` to null-safety. Needed to migrate `shared_preferences`.

It contributes to fixing https://github.com/flutter/flutter/issues/74876.

## Pre-launch Checklist

- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: packages/path_provider/path_provider_linux/lib/path_provider_linux.dart [hunk 1]
```diff
@@ -36,11 +36,11 @@ class PathProviderLinux extends PathProviderPlatform {
 
   @override
   Future<String> getApplicationDocumentsPath() {
-    return Future.value(xdg.getUserDirectory('DOCUMENTS').path);
+    return Future.value(xdg.getUserDirectory('DOCUMENTS')!.path);
```
Comment (gaetschwartz) [inline packages/path_provider/path_provider_linux/lib/path_provider_linux.dart:39]: I am unsure if we can assume `xdg.getUserDirectory('DOCUMENTS')` is null only if the directory is incorrect thus forcing it wouldn't be an issue or not ?

**Comment by gaetschwartz** [comment]
I'm unsure as to why the CI is failing, it's related to path_provider_windows, which I didn't touch.

**Comment by stuartmorgan** [comment]
There's an existing PR for this at #3330; please comment there if you have changes to suggest.

**Comment by gaetschwartz** [comment]
@stuartmorgan What is wrong with the other PR ? It's currently blocking the migration of shared_preferences, one of the most used package of this repo. Getting it reviewed would help a lot :)

**Comment by stuartmorgan** [comment]
> @stuartmorgan What is wrong with the other PR ?

I'm not sure what you are asking here. Why do you think something is wrong with it?

> It's currently blocking the migration of shared_preferences, one of the most used package of this repo. Getting it reviewed would help a lot :)

I expect to review it next week. It's only actually been reviewable for about a day, since a core dependency hadn't been migrated.

**Comment by gaetschwartz** [comment]
@stuartmorgan Alright, was just wondering why it hadn't been reviewed. Missed the part where it needed a code dependency 😊 Sorry if it sounded weird, wasn't the point 💙☺️

**Pull request closed without merging**